src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -388,6 +432,42 @@ private JsonRpcRequest ReadRequest(JToken json)
                 args is JArray argsArray ? (object)PartiallyParsePositionalArguments(argsArray) :
                 null;
 
+            // If method is $/progress, get the progress instance from the dictionary and call Report
+            string method = json.Value<string>("method");
+
+            if (string.Equals(method, ProgressRequestSpecialMethod))
+            {
+                try
+                {
+                    JToken progressId =
+                        args is JObject ? args["token"] :
+                        args is JArray ? args[0] :
+                        null;
+
+                    JToken value =
+                        args is JObject ? args["value"] :
+                        args is JArray ? args[1] :
+                        null;
+
+                    lock (this.progressMap)
+                    {
+
+                        if (this.progressMap.TryGetValue(progressId.Value<long>(), out object progress))
+                        {
+                            Type iprogressOfTType = MessageFormatterHelper.FindIProgressOfT(progress.GetType());
+                            Type valueType = iprogressOfTType.GenericTypeArguments[0];
+                            var reportMethod = iprogressOfTType.GetRuntimeMethod(nameof(IProgress<int>.Report), new Type[] { valueType });
+                            object typedValue = value.ToObject(valueType);
+                            reportMethod.Invoke(progress, new object[] { typedValue });
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    this.Rpc.TraceSource.TraceEvent(TraceEventType.Warning, (int)JsonRpc.TraceEvents.ProgressNotificationError, e.Message);
```

Comment:
>e.Message [](start = 128, length = 9)

tip: Just pass `e` here. That will call an overload that just provides a data object rather than a string, and the entire exception object can be far more meaningful when debugging or looking at logs than just the message.